src/js/containers/search/visualizations/rank/RankVisualizationWrapperContainer.jsx

```diff
@@ -37,351 +35,336 @@ const propTypes = {
-                        subaward={this.props.subaward}
-                        isDefCodeInFilter={this.props.reduxFilters?.defCodes?.counts}
-                        togglePicker={this.togglePicker} />
+                        {...childProps}
+                        changeScope={changeScope}
+                        nextPage={nextPage}
+                        previousPage={previousPage}
+                        subaward={props.subaward}
+                        isDefCodeInFilter={props.reduxFilters?.defCodes?.counts}
+                        togglePicker={togglePicker} />
                 );
             case 'industryCode':
                 return (
                     <SpendingByIndustryCodeSection
-                        {...this.state}
-                        changeScope={this.changeScope}
-                        nextPage={this.nextPage}
-                        previousPage={this.previousPage}
-                        industryCodeError={this.props.subaward}
-                        subaward={this.props.subaward}
-                        isDefCodeInFilter={this.props.reduxFilters?.defCodes?.counts}
-                        togglePicker={this.togglePicker} />
+                        {...childProps}
+                        changeScope={changeScope}
+                        nextPage={nextPage}
+                        previousPage={previousPage}
+                        industryCodeError={props.subaward}
+                        subaward={props.subaward}
+                        isDefCodeInFilter={props.reduxFilters?.defCodes?.counts}
+                        togglePicker={togglePicker} />
                 );
             default:
                 return (
                     <SpendingByAgencySection
-                        {...this.state}
-                        changeScope={this.changeScope}
-                        nextPage={this.nextPage}
-                        previousPage={this.previousPage}
+                        {...childProps}
+                        changeScope={changeScope}
+                        nextPage={nextPage}
+                        previousPage={previousPage}
                         agencyType="awarding"
-                        subaward={this.props.subaward}
-                        isDefCodeInFilter={this.props.reduxFilters?.defCodes?.counts}
-                        togglePicker={this.togglePicker} />
+                        subaward={props.subaward}
+                        isDefCodeInFilter={props.reduxFilters?.defCodes?.counts}
+                        togglePicker={togglePicker} />
                 );
         }
-    }
-
-    changeSpendingBy(spendingBy) {
-        this.setState({
-            spendingBy,
-            scope: defaultScopes[spendingBy]
-        });
-    }
+    };
+
+    useEffect(() => {
+        // fetch data when scope or page changes
+        fetchData();
+        /* eslint-disable-next-line react-hooks/exhaustive-deps */
+    }, [page, scope]);
```

Comment:
did it add these before you put the eslint line there?

Reply:
no i added them